### PR TITLE
Fix custom hit testing

### PR DESCRIPTION
**Pull request by wieslawsoltes**
## What does the pull request do?

Fixes custom hit-testing broken in https://github.com/AvaloniaUI/Avalonia/pull/5923

## What is the current behavior?

Custom hit testing when custom title bar is used not working.

## What is the updated/expected behavior with this PR?

Fixes custom hit-testing.

## How was the solution implemented (if it's not obvious)?

Check for correct interface.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

File: src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs [hunk 1]
```diff
@@ -310,7 +310,7 @@ private bool FilterAndClip(IVisualNode node, ref Rect? clip)
                         clipped = !node.GeometryClip.FillContains(controlPoint.Value);
                     }
 
-                    if (!clipped && node.Visual is ICustomHitTest custom)
+                    if (!clipped && node.Visual is ICustomSimpleHitTest custom)
```
Comment (grokys) [inline src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs:313]: `ICustomSimpleHitTest` only [applies to the immediate renderer](https://github.com/AvaloniaUI/Avalonia/blob/ef38e9dd3a25e513f449a51ef09508af65140841/src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs#L8-L10) and `SceneGraph` is a deferred renderer concept, so I don't think this change is correct. Why was it needed?

File: src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public interface ICustomHitTest : ICustomSimpleHitTest
     public static class CustomSimpleHitTestExtensions
     {
         public static bool HitTestCustom(this IVisual visual, Point point)
-            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? visual.TransformedBounds?.Contains(point) == true;
+            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? false;
```
Comment (grokys) [inline src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs:28]: In the absence of `ICustomSimpleHitTest` we still need to check against the control bounds I think?

**Review by grokys — CHANGES_REQUESTED** [review]

File: src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs (hunk 1 above)
Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs]: Ah yeah not really necessary here, reverted

File: src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs (hunk 2 above)
Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs]: Reverted but switched to using Bounds instead of TransformedBounds

File: src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public interface ICustomHitTest : ICustomSimpleHitTest
     public static class CustomSimpleHitTestExtensions
     {
         public static bool HitTestCustom(this IVisual visual, Point point)
-            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? visual.TransformedBounds?.Contains(point) == true;
+            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? visual.Bounds.Contains(point);
```
Comment (grokys) [inline src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs:28]: Why the change to `Bounds` here? The documentation for `ICustomSimpleHitTest.HitTest` says: "The point to hit test in _global coordinate space_". `Bounds` is in the control's coordinate space, so shouldn't this stay as `TransformedBounds`?

**Review by grokys — CHANGES_REQUESTED** [review]
Still not sure this is right, and I've been unable to test it: the description says "Custom hit testing when custom title bar is used not working." but when I run ControlCatalog and go to "Dialogs -> Decorated Window" the custom title bar seems to work fine?

First of all how do I repro the problem in order to verify the fix?

**Comment by wieslawsoltes** [comment]
> Still not sure this is right, and I've been unable to test it: the description says "Custom hit testing when custom title bar is used not working." but when I run ControlCatalog and go to "Dialogs -> Decorated Window" the custom title bar seems to work fine?
> 
> First of all how do I repro the problem in order to verify the fix?

@grokys repro:
1. Run `ControlCatalog.NetCore`
2. Got to `Window Customizations` page
3. Check `Extend Client Area to Decorations` and `Prefer System Chrome`
4. Try dragging window by titlebar

ControlCatalog NetCore_lArfFfokJP

File: src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs (hunk 3 above)
Comment (wieslawsoltes) [reply src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs]: This change made it work, probably something else is happening, but could not find the source.

Comment (grokys) [reply src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs:28]: This change doesn't change anything in your repro as far as I can tell: 

1. With deferred renderer, with and without this change it works
2. With immediate renderer, with and without this change it doesn't work

And give that this code should only apply to immediate renderer - the problem is elswhere. Pretty sure this should be `TransformedBounds`.
is elsewhere.

Comment (grokys) [reply src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs:28]: Custom hit-testing doesn't seem to have been working with immediate renderer even before #5923.

**Review by grokys — APPROVED** [review]
Ok, reverted the `TransformedBounds` change as I've checked and this PR fixes custom title bar hit-testing on the deferred renderer without it and it was already broken on the immediate renderer before #5923, so makes no difference there either.
